receiver/sqlserverreceiver/scraper.go

```diff
@@ -352,6 +353,14 @@ func (s *sqlServerScraperHelper) recordDatabasePerfCounterMetrics(ctx context.Co
 			} else {
 				s.mb.RecordSqlserverLockTimeoutRateDataPoint(now, val)
 			}
+		case lockWaitCount:
+			val, err := retrieveInt(row, valueKey)
```

Comment:
`retrieveInt` is called here instead of `strconv.ParseInt` conform to new functionality introduced in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39905 (which hasn't been merged yet).